server/src/test/java/org/elasticsearch/rest/action/admin/indices/RestGetMappingActionTests.java

```diff
@@ -69,4 +69,20 @@ public void testTypeInPath() {
         assertEquals(1, channel.errors().get());
         assertEquals(RestStatus.BAD_REQUEST, channel.capturedResponse().status());
     }
+
+    public void testTypeUrlParamerterDeprecation() throws Exception {
+        Map<String, String> params = new HashMap<>();
+        params.put(INCLUDE_TYPE_NAME_PARAMETER, "true");
+        RestRequest request = new FakeRestRequest.Builder(xContentRegistry())
+            .withMethod(RestRequest.Method.GET)
+            .withParams(params)
+            .withPath("some_index/some_type/_mapping/some_field")
+            .build();
+
+        RestGetMappingAction handler = new RestGetMappingAction(Settings.EMPTY, mock(RestController.class));
+        handler.prepareRequest(request, mock(NodeClient.class));
```

Comment:
This doesn't work without some more involved mocking which I think isn't necessary here. If using the "standard" approach from other tests I get a NPE in `org.elasticsearch.rest.action.admin.indices.RestGetMappingAction.lambda$0(RestGetMappingAction.java:105` because the client isn't fully mocked out. I'd rather keep it this way, it tests what it is supposed to test.